flo-workflow/src/main/java/com/spotify/flo/context/InstrumentedContext.java

```diff
@@ -129,17 +129,11 @@ public static EvalContext composeWith(EvalContext baseContext, Listener listener
 
   @Override
   public <T> Value<T> invokeProcessFn(TaskId taskId, Fn<T> processFn) {
-    return delegate.invokeProcessFn(taskId, () -> {
-      listener.status(taskId, Listener.Phase.START);
-      try {
-        final T value = processFn.get();
-        listener.status(taskId, Listener.Phase.SUCCESS);
-        return value;
-      } catch (Throwable t) {
```

Comment:
Why there is no try-catch anymore? Is it safe to call `invokeProcessFn`?

Reply:
The try-catch was around `processFn` and with the change we don't call `get` on it directly and let `value.consume/onFail` handle it.

Reply:
Gets propagated all the way up and that fails evaluation of the task, I think. Can check it later.

Reply:
Calling `invokeProcessFn` is "safe" as in if it throws an unchecked exception that is either a jvm failure or a flo internal failure and we let that failure propagate all the way to the program entry point, i.e. "panic" and crash. The call to `invokeProcessFn` will not fail due to exceptions thrown by user code.